cassandra/cython_marshal.pyx

```diff
@@ -0,0 +1,221 @@
+    cdef char *p = <char *> &x
+    swap_order(<char *> &x, 4)
+    return x
+
+cdef inline bytes int16_pack(int16_t x):
+    return pack(<char *> &x, 2)
+
+cdef inline int16_t int16_unpack(Buffer *buf) except ?0xDED:
+    cdef int16_t x = (<int16_t *> buf_read(buf, 2))[0]
+    swap_order(<char *> &x, 2)
+    return x
+
+cdef inline bytes int8_pack(int8_t x):
+    return (<char *> &x)[:1]
+
+cdef inline int8_t int8_unpack(Buffer *buf) except ?80:
+    return (<int8_t *> buf_read(buf, 1))[0]
+
+cdef inline bytes uint64_pack(uint64_t x):
+    return pack(<char *> &x, 8)
+
+cdef inline uint64_t uint64_unpack(Buffer *buf) except ?0xDEAD:
+    cdef uint64_t x = (<uint64_t *> buf_read(buf, 8))[0]
+    swap_order(<char *> &x, 8)
+    return x
+
+cdef inline bytes uint32_pack(uint32_t x):
+    return pack(<char *> &x, 4)
+
+cdef inline uint32_t uint32_unpack(Buffer *buf) except ?0xDEAD:
+    cdef uint32_t x = (<uint32_t *> buf_read(buf, 4))[0]
+    swap_order(<char *> &x, 4)
+    return x
+
+cdef inline bytes uint16_pack(uint16_t x):
+    return pack(<char *> &x, 2)
+
+cdef inline uint16_t uint16_unpack(Buffer *buf) except ?0xDEAD:
+    cdef uint16_t x = (<uint16_t *> buf_read(buf, 2))[0]
+    swap_order(<char *> &x, 2)
+    return x
+
+cdef inline bytes uint8_pack(uint8_t x):
+    return pack(<char *> &x, 1)
+
+cdef inline uint8_t uint8_unpack(Buffer *buf) except ?0xff:
+    return (<uint8_t *> buf_read(buf, 1))[0]
+
+cdef inline bytes double_pack(double x):
+    return pack(<char *> &x, 8)
+
+cdef inline double double_unpack(Buffer *buf) except ?1.74:
+    cdef double x = (<double *> buf_read(buf, 8))[0]
+    swap_order(<char *> &x, 8)
+    return x
+
+cdef inline bytes float_pack(float x):
+    return pack(<char *> &x, 4)
+
+cdef inline float float_unpack(Buffer *buf) except ?1.74:
+    cdef float x = (<float *> buf_read(buf, 4))[0]
+    swap_order(<char *> &x, 4)
+    return x
+
+# int64_pack, int64_unpack = _make_packer('>q')
+# int32_pack, int32_unpack = _make_packer('>i')
+# int16_pack, int16_unpack = _make_packer('>h')
+# int8_pack, int8_unpack = _make_packer('>b')
+# uint64_pack, uint64_unpack = _make_packer('>Q')
+# uint32_pack, uint32_unpack = _make_packer('>I')
+# uint16_pack, uint16_unpack = _make_packer('>H')
+# uint8_pack, uint8_unpack = _make_packer('>B')
+# float_pack, float_unpack = _make_packer('>f')
+# double_pack, double_unpack = _make_packer('>d')
+
+# Special case for cassandra header
+header_struct = struct.Struct('>BBbB')
+header_pack = header_struct.pack
+header_unpack = header_struct.unpack
+
+# in protocol version 3 and higher, the stream ID is two bytes
+v3_header_struct = struct.Struct('>BBhB')
+v3_header_pack = v3_header_struct.pack
+v3_header_unpack = v3_header_struct.unpack
+
+
+cdef varint_unpack(term):
+    """Unpack a variable-sized integer"""
+    if PY3:
+        return varint_unpack_py3(term)
+    else:
+        return varint_unpack_py2(term)
+
+# TODO: Optimize these two functions
```

Comment:
The trouble here is that static function calls are faster as they don't have to build a argument tuple. I'll make them cdef functions to illustrate this. Thanks!